src/libs/barcode/lv_barcode.c

```diff
@@ -94,16 +95,23 @@ void lv_barcode_set_direction(lv_obj_t * obj, lv_dir_t direction)
     barcode->direction = direction;
 }
 
-lv_result_t lv_barcode_update(lv_obj_t * obj, const char * data)
+void lv_barcode_set_tiled(lv_obj_t * obj, bool tiled)
 {
     LV_ASSERT_OBJ(obj, MY_CLASS);
-    LV_ASSERT_NULL(data);
 
-    lv_result_t res = LV_RESULT_INVALID;
     lv_barcode_t * barcode = (lv_barcode_t *)obj;
+    barcode->tiled = tiled;
+    lv_image_set_inner_align(obj, tiled ? LV_IMAGE_ALIGN_TILE : LV_IMAGE_ALIGN_DEFAULT);
+}
+
+lv_result_t lv_barcode_update(lv_obj_t * obj, const char * data)
+{
+    LV_ASSERT_OBJ(obj, MY_CLASS);
+    LV_ASSERT_NULL(data);
 
     if(data == NULL || lv_strlen(data) == 0) {
         LV_LOG_WARN("data is empty");
+        lv_barcode_clear(obj);
         return LV_RESULT_INVALID;
     }
```

Comment:
Are we sure this should warn? The test case uses `NULL` `data` it to clear it.

Reply:
On second thought, removing `lv_barcode_update(obj, NULL)` as a way of clearing barcode should alert users when they pass in incorrect empty data, as this can easily happen.